File: Medium/1481. Least Number of Unique Integers after K Removals/sol.js
var findLeastNumOfUniqueInts = function(arr, k) {
    let mp = new Map();
    let freq = new MinPriorityQueue();

    for(let i of arr) {
        mp.set(i, mp.get(i) + 1 || 1);
    }

    for(let [key, value] of mp) {
        freq.enqueue(value);
    }

    while(!freq.isEmpty()) {
        let i = freq.front().element;
        if(k >= i) {
            k -= i;
            freq.dequeue();
        } else break;
    }

    return freq.size();
};

File: Medium/1481. Least Number of Unique Integers after K Removals/sol.java
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        Map<Integer, Integer> mp = new HashMap<>();
        PriorityQueue<Integer> freq = new PriorityQueue<>();

        for(int i : arr) {
            mp.put(i, mp.getOrDefault(i, 0) + 1);
        }

        for(int i : mp.values()) freq.add(i);

        while(!freq.isEmpty()) {
            int i = freq.peek();
            if(k >= i) {
                k -= i;
                freq.poll();
            } else break;
        }

        return freq.size();
    }
}